employee-rostering-backend/src/test/java/org/optaweb/employeerostering/skill/SkillServiceUnitTest.java

```diff
@@ -0,0 +1,270 @@
+/*
+ * Copyright 2019 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaweb.employeerostering.skill;
+
+import javax.transaction.Transactional;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import org.optaweb.employeerostering.domain.skill.Skill;
+import org.optaweb.employeerostering.service.SkillService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.http.MediaType;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.request.MockMvcRequestBuilders;
+import org.springframework.test.web.servlet.result.MockMvcResultMatchers;
+import org.springframework.web.util.NestedServletException;
+
+import static org.assertj.core.api.Assertions.assertThatExceptionOfType;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+@RunWith(SpringRunner.class)
+@SpringBootTest
+@AutoConfigureMockMvc
+@Transactional
+public class SkillServiceUnitTest {
```

Comment:
The "Unit" in the test name is superfluous. "IT" as a class name suffix is required by Maven Failsafe plugin. It's used to exclude integration test (usually long-running) from the `test` phase and defer them to `integration-test` phase which is executed less often during development.

As I'm writing this, I realize that your `*ControllerIT` tests are actually not executed at all because the project doesn't use Failsafe plugin and Surefire's default include pattern doesn't match ITs. So unless you want to execute them in `integration-test` phase (I don't see a reason for that at the moment), rename them to `*ControllerTest`, please.

And please remove "Unit" from test names as well. There is no such convention yet and for consistency reasons, let's not introduce it. It would be hard to follow such convention consistently. We'd have to think about classification of tests as unit, component, integration, something in between? and reflect it in class names. I don't think we want to do that.

Reply:
Wow, I didn't know something as subtle as a test class name could affect test phases. Will fix this and test name convention in next push. Thanks Jiri!